src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -219,6 +225,23 @@ public final void apply(Project project) {
         // (but that's internal)
         project.getPluginManager().apply("java-base");
 
+        // Create "platform" configuration in root project, which will hold the strictConstraints
+        NamedDomainObjectProvider<Configuration> gcvLocksConfiguration = project.getConfigurations()
+                .register("gcvLocks", conf -> {
+                    conf.getAttributes().attribute(Usage.USAGE_ATTRIBUTE, internalUsage);
+                    conf.getOutgoing().capability(GCV_LOCKS_CAPABILITY);
+                    conf.setCanBeResolved(false);
+                    conf.setVisible(false);
+                });
+
+        ProjectDependency locksDependency =
+                (ProjectDependency) project.getDependencies().create(project);
+        locksDependency.capabilities(moduleDependencyCapabilitiesHandler ->
```

Comment:
is this necessary given that we already have configured the configuration?

Reply:
yes - we will resolve multiple variants of the same project (root project) so if they don't have different (i.e. explicit and different) capabilities, they will conflict.

Reply:
oh right and you can't depend on an explicit capability just with attributes, you need to specify the capability.

If anything, we might not need to specify the attribute directly, but it's a bit more clear this way I think.